core/shared/platform/common/posix/posix_socket.c

```diff
@@ -176,3 +176,50 @@ os_socket_inet_network(const char *cp, uint32 *out)
     *out = ntohl(inet_addr(cp));
     return BHT_OK;
 }
+
+int
+os_socket_addr_local(bh_socket_t socket, uint8_t *buf, size_t buflen,
+                     uint16_t *port, uint8_t *is_ipv4)
+{
+    struct sockaddr_storage addr_storage = { 0 };
+    socklen_t addr_len = sizeof(addr_storage);
+    int ret;
+
+    assert(buf);
+    assert(is_ipv4);
+    assert(port);
+
+    ret = getsockname(socket, (struct sockaddr *)&addr_storage, &addr_len);
+
+    if (ret != BHT_OK) {
+        return BHT_ERROR;
+    }
+
+    switch (addr_storage.ss_family) {
+        case AF_INET:
+        {
+            struct sockaddr_in *addr_in = (struct sockaddr_in *)&addr_storage;
+
+            assert(buflen >= sizeof(addr_in->sin_addr));
+            *port = ntohs(addr_in->sin_port);
+            memcpy(buf, &addr_in->sin_addr, sizeof(addr_in->sin_addr));
+            *is_ipv4 = true;
+        } break;
```

Comment:
```suggestion
          break;
        } 
```

Reply:
Hmm, that seems to be automatically applied by clang-format-12. Also, I see it's the same in other files (e.g. [this one](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/samples/wasm-c-api/src/callback.c#L15))

Reply:
```suggestion
            break;
        }
```